src/ut/http_request_test.cpp

```diff
@@ -0,0 +1,201 @@
+/**
+ * @file http_request_test.cpp UT for the HttpRequest interface.
+ *
+ * Copyright (C) Metaswitch Networks 2018
+ * If license terms are provided to you in a COPYING file in the root directory
+ * of the source code repository by which you are accessing this code, then
+ * the license outlined in that COPYING file applies to your use.
+ * Otherwise no rights are granted except for those provided to you by
+ * Metaswitch Networks in a separate written agreement.
+ */
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "mock_httpclient.h"
+#include "http_request.h"
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::DoAll;
+using ::testing::IsEmpty;
+using ::testing::Pointee;
+using ::testing::SetArgReferee;
+using ::testing::SetArgPointee;
+
+
+class HttpRequestTest : public ::testing::Test
+{
+
+  HttpRequestTest()
+  {
+    _client = new MockHttpClient();
+  }
+
+  ~HttpRequestTest()
+  {
+    delete _client; _client = NULL;
+  }
+
+  MockHttpClient* _client;
+};
+
+// Build and send a basic http request, with the default parameters
+// The send method is passed through, so we shouldn't need individual tests to
+// cover all of POST, PUT, GET, DELETE.
+TEST_F(HttpRequestTest, send_basic_request)
```

Comment:
In general, most of our tests are CamelCase, so this would be `SendBasicRequest`. Same applies to all the below